src/lib/mongoBackend/mongoGetSubscriptions.cpp

```diff
@@ -130,46 +130,94 @@ static void setSubject(Subscription* s, const BSONObj& r)
+static void setNotification(Subscription* subP, const BSONObj& r, const std::string& tenant)
 {
   // Attributes
   std::vector<BSONElement> attrs = getFieldF(r, CSUB_ATTRS).Array();
   for (unsigned int ix = 0; ix < attrs.size(); ++ix)
   {
     std::string attr = attrs[ix].String();
 
-    s->notification.attributes.push_back(attr);
+    subP->notification.attributes.push_back(attr);
   }
 
-  s->notification.httpInfo.url     = getStringFieldF(r, CSUB_REFERENCE);
-  s->throttling                    = r.hasField(CSUB_THROTTLING)?       getIntOrLongFieldAsLongF(r, CSUB_THROTTLING)       : -1;
-  s->notification.lastNotification = r.hasField(CSUB_LASTNOTIFICATION)? getIntOrLongFieldAsLongF(r, CSUB_LASTNOTIFICATION) : -1;
-  s->notification.timesSent        = r.hasField(CSUB_COUNT)?            getIntOrLongFieldAsLongF(r, CSUB_COUNT)            : -1;
+  //
+  // Http Info
+  //
+  // FIXME P5: qs and header population code also in mongoSubCache.cpp AND mongoUpdateSubscription.cpp.
+  //           Maybe it makes sense to define common functions
+  //           in the HttpInfo for populating fields from BSONObj
+  //
+  subP->notification.httpInfo.url      = getStringFieldF(r, CSUB_REFERENCE);
+  subP->notification.httpInfo.extended = r.hasField(CSUB_EXTENDED)? getBoolFieldF(r, CSUB_EXTENDED)  : false;
+
+  if (subP->notification.httpInfo.extended)
+  {
+    subP->notification.httpInfo.payload  = r.hasField(CSUB_PAYLOAD)? getStringFieldF(r, CSUB_PAYLOAD) : "";
+
+    if (r.hasField(CSUB_METHOD))
+    {
+      subP->notification.httpInfo.verb = str2Verb(getFieldF(r, CSUB_METHOD).String());
+    }
+
+    // qs
+    if (r.hasField(CSUB_QS))
+    {
+      BSONObj qs = getFieldF(r, CSUB_QS).Obj();
+
+      for (BSONObj::iterator i = qs.begin(); i.more();)
+      {
+        BSONElement e = i.next();
+        subP->notification.httpInfo.qs[e.fieldName()] = e.String();
+      }
+    }
+
+    // headers
+    if (r.hasField(CSUB_HEADERS))
+    {
+      BSONObj headers = getFieldF(r, CSUB_HEADERS).Obj();
+
+      for (BSONObj::iterator i = headers.begin(); i.more();)
+      {
+        BSONElement e = i.next();
+        subP->notification.httpInfo.headers[e.fieldName()] = e.String();
+      }
+    }
+  }
+
+  subP->throttling                    = r.hasField(CSUB_THROTTLING)?       getIntOrLongFieldAsLongF(r, CSUB_THROTTLING)       : -1;
+  subP->notification.lastNotification = r.hasField(CSUB_LASTNOTIFICATION)? getIntOrLongFieldAsLongF(r, CSUB_LASTNOTIFICATION) : -1;
+  subP->notification.timesSent        = r.hasField(CSUB_COUNT)?            getIntOrLongFieldAsLongF(r, CSUB_COUNT)            : -1;
+
 
   //
   // Check values from subscription cache, update object from cache-values if necessary
   //
-  CachedSubscription* cSubP = subCacheItemLookup(tenant.c_str(), s->id.c_str());
+  // NOTE: only 'lastNotificationTime' and 'count'
+  //
+  cacheSemTake(__FUNCTION__, "get lastNotification and count");
```

Comment:
I think this code was last modified before Orion 1.1 (current last version).

However, to be sure you can use "Blame" on that file (in develop branch) to know in which commit (and its associated date) that part of the code was last modified.